src/modules/workspaces/workspaces.usecase.spec.ts

```diff
@@ -949,4 +950,441 @@ describe('WorkspacesUsecases', () => {
       expect(assignableSpace).toBe(BigInt(300000));
     });
   });
+
+  describe('teams', () => {
+    describe('createTeam', () => {
+      it('When workspace is not found, then fail', async () => {
+        const user = newUser();
+        const workspace = newWorkspace({ owner: user });
+
+        jest.spyOn(workspaceRepository, 'findOne').mockResolvedValue(null);
+        await expect(
+          service.createTeam(user, workspace.id, {
+            name: 'test',
+            managerId: '',
+          }),
+        ).rejects.toThrow(BadRequestException);
+      });
+
+      it('When maximum teams in a workspace is reached, then throw', async () => {
+        const user = newUser();
+        const workspace = newWorkspace({ owner: user });
+
+        jest.spyOn(workspaceRepository, 'findOne').mockResolvedValue(workspace);
+        jest
+          .spyOn(teamRepository, 'getTeamsInWorkspaceCount')
+          .mockResolvedValue(10);
+
+        await expect(
+          service.createTeam(user, workspace.id, {
+            name: 'test',
+            managerId: '',
+          }),
+        ).rejects.toThrow(BadRequestException);
+      });
+
+      it('When maximum teams in a workspace is still not reached, then create workspace', async () => {
```

Comment:
In the workspace lifecycle, a workspace comes first. How is it possible to check the maximum number of teams in a workspace of a non-existent workspace?

I assume that the legend is wrong by reading the code and it should finish with _then the team is created succesfully_